### DISCUSSION TITLE
add a parameter about the slot in seurat object

### DISCUSSION THREAD

**Discussion by liuyu988** [satijalab/seurat#5382] [Q&A]
**Dear sir,could the seurat object add a slot parameter,except data,counts,scale.data?thanks.**

ERROR:
Error in (function (cl, name, valueClass)  : 
  ‘chip’ is not a slot in class “Assay”

**Comment by liuyu988** [comment]
I want to add a matrix to a seurat object
my code:
> setAssayData(x,slot="add matrix",matrix)
Error in match.arg(arg = slot):
 'arg' should be one of "data"\"scale.data"\"counts"

**Comment by samuel-marsh** [comment]
Hi,

What is the matrix that you are trying to add to the object?  What is the goal?

Best,
Sam

**Comment by liuyu988** [comment]
Hi,

 I want to add a matrix that rows are gene and cols are cells to the seurat object, and I want to add it to the slot, however , the slot may have only several fixed objects by reading your source codes in github. So I wonder if there is any method to achieve this thinking.

Best!
Yu Liu

发件人：Samuel Marsh ***@***.***>
发送日期：2021-12-13 23:42:29
收件人：satijalab/seurat ***@***.***>
抄送人：liuyu988 ***@***.***>,Author ***@***.***>
主题：Re: [satijalab/seurat] add a parameter about the slot in seurat object (Discussion #5382)

Hi,
What is the matrix that you are trying to add to the object? What is the goal?
Best,
 Sam
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by samuel-marsh** [comment]
What is in the matrix?  How is it different from whatever data is already in object?

**Comment by liuyu988** [comment]
It is a chip data matrix

发件人：Samuel Marsh ***@***.***>
发送日期：2021-12-16 10:59:08
收件人：satijalab/seurat ***@***.***>
抄送人：liuyu988 ***@***.***>,Author ***@***.***>
主题：Re: [satijalab/seurat] add a parameter about the slot in seurat object (Discussion #5382)

What is in the matrix? How is it different from whatever data is already in object?
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by samuel-marsh** [comment]
What is the goal of adding it to the Seurat object?  Analysis?  Or just trying to store it with other data?
